cscs-checks/analytics/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,109 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+@rfm.simple_test
+class HPCGCheckRef(rfm.RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.descr = 'HPCG reference benchmark'
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['craype-hugepages8M']
+        self.build_system = 'Make'
+        self.build_system.options = ['arch=MPI_GCC_OMP']
```

Comment:
Does it make sense to add the compilation flags here, per system?
```
self.build_system.cxxflags = '-xCORE-AVX2 -qopenmp -std=c++11 $(HPCG_DEFS)'
```

Because not all systems are AVX2. Again it only makes sense if we extend the test systems.

Reply:
Let's focus on Daint/Dom right now. You could open an issue for extending this test on Fulen.